float32 df_1
float32 df_2
float32 far_good_angle
float32 dd_1
float32 dd_2
float32 good_angle
float32 final_angle
float32 af_angle
